### PR TITLE
Upgrade to 3.0.6 (Vert.x 3.x)

### PR DESCRIPTION
**Pull request by aaloise**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -17,7 +17,7 @@
   <properties>
     <stack.version>3.9.3-SNAPSHOT</stack.version>
     <asciidoc.dir>${project.basedir}/src/main/asciidoc</asciidoc.dir>
-    <groovy.version>3.0.2</groovy.version>
+    <groovy.version>3.0.4</groovy.version>
```
Comment (vietj) [inline pom.xml:20]: Why not 3.0.5 ?

Comment (aaloise) [reply pom.xml:20]: 3.0.5 is throwing a ConcurrentModificationException at VertxTransformation. This wasn't happening until 3.0.4, so I guess it is a bug. I raised an issue about this at Groovy, expecting for a solution.

Comment (aaloise) [reply pom.xml:20]: Or maybe an workaround

**Comment by aaloise** [comment]
As explained, build was failing due to a modification in Groovy 3.0.5 regarding AST Transformation. An issue was raised in Groovy (https://issues.apache.org/jira/browse/GROOVY-9666), so the Groovy team can investigate what they will do about it.

Meanwhile, I manage to get an workaround. Branch 3.9.x now build successfully whit 3.0.5 version of Groovy.

@vietj Could you check if is ok under your perspective? If it is, I think is safe to merge.

File: pom.xml (hunk 1 above)
Comment (aaloise) [reply pom.xml]: It builds now with 3.0.5

**Pull request merged**